Wikipedia/Code/WMFCVLInfo.m

```diff
@@ -0,0 +1,337 @@
+    newSize.height = 0;
+    [self enumerateColumnsWithBlock:^(WMFCVLColumn * _Nonnull column, NSUInteger idx, BOOL * _Nonnull stop) {
+        CGFloat columnHeight = column.height;
+        if (columnHeight > newSize.height) {
+            newSize.height = columnHeight;
+        }
+    }];
+    self.contentSize = newSize;
+}
+
+- (void)updateContentSizeWithInvalidationContext:(WMFCVLInvalidationContext *)context {
+    CGSize oldContentSize = self.contentSize;
+    
+    [self updateContentSize];
+    
+    CGSize contentSizeAdjustment = CGSizeMake(self.contentSize.width - oldContentSize.width, self.contentSize.height - oldContentSize.height);
+    context.contentSizeAdjustment = contentSizeAdjustment;
+}
+
+- (void)updateWithInvalidationContext:(nonnull WMFCVLInvalidationContext *)context delegate:(id <WMFColumnarCollectionViewLayoutDelegate>)delegate collectionView:(UICollectionView *)collectionView {
+    if (context.boundsDidChange) {
+        [self resetSections];
+        [self layoutForBoundsSize:context.newBounds.size withDelegate:delegate collectionView:collectionView invalidationContext:context];
+    } else if (context.originalLayoutAttributes && context.preferredLayoutAttributes) {
+        UICollectionViewLayoutAttributes *originalAttributes = context.originalLayoutAttributes;
+        UICollectionViewLayoutAttributes *preferredAttributes = context.preferredLayoutAttributes;
+        NSIndexPath *indexPath = originalAttributes.indexPath;
+        
+        WMFCVLSection *invalidatedSection = self.sections[indexPath.section];
+        WMFCVLColumn *invalidatedColumn = invalidatedSection.column;
+        
+        CGSize sizeToSet = preferredAttributes.frame.size;
+        sizeToSet.width = invalidatedColumn.width;
+        
+        if (originalAttributes.representedElementCategory == UICollectionElementCategoryCell) {
+            [invalidatedColumn setSize:sizeToSet forItemAtIndexPath:indexPath invalidationContext:context];
+        } else if ([originalAttributes.representedElementKind isEqualToString:UICollectionElementKindSectionHeader]) {
+            [invalidatedColumn setSize:sizeToSet forHeaderAtIndexPath:indexPath invalidationContext:context];
+        } else if ([originalAttributes.representedElementKind isEqualToString:UICollectionElementKindSectionFooter]) {
+            [invalidatedColumn setSize:sizeToSet forFooterAtIndexPath:indexPath invalidationContext:context];
+        }
+        [self updateContentSizeWithInvalidationContext:context];
+        if (self.metrics.numberOfColumns == 1 && originalAttributes.frame.origin.y < collectionView.contentOffset.y) {
+            context.contentOffsetAdjustment = CGPointMake(0, context.contentSizeAdjustment.height);
+        }
+    }
+}
+
+- (void)layoutForBoundsSize:(CGSize)size withDelegate:(id <WMFColumnarCollectionViewLayoutDelegate>)delegate collectionView:(UICollectionView *)collectionView invalidationContext:(nullable WMFCVLInvalidationContext *)context {
```

Comment:
This is a pretty long method - which I know this has all the calculation logic so there might not be a way around it. 

It would be nice to maybe break it up into some methods if possible. Or maybe just some light sprinkling of high level comments to help with scannability when reading through it.